File: src/main/java/com/magina/resource/service/CloudDiskService.java
package com.magina.resource.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import org.springframework.util.StringUtils;

import com.magina.resource.domain.CloudDisk;
import com.magina.resource.repository.CloudDiskRepository;

@Service
public class CloudDiskService {
    
    @Autowired
    private CloudDiskRepository cloudDiskRepo;
    
    public Page<CloudDisk> listLinks(Pageable pageable, String query) {
        return StringUtils.hasText(query) ? cloudDiskRepo.findByNameLike(query, pageable) : cloudDiskRepo.findAll(pageable);
    }
    
    public CloudDisk getLink(Long id) {
        return cloudDiskRepo.findOne(id);
    }
    
    @Transactional
    public void delete(Long id) {
        cloudDiskRepo.delete(id);
    }
    
    @Transactional
    public void save(CloudDisk cloudDisk) {
        cloudDiskRepo.save(cloudDisk);
    }

}

File: src/main/kotlin/com/magina/resource/service/CloudDiskService.kt
package com.magina.resource.service

import com.magina.resource.dao.CloudDiskDAO
import com.magina.resource.domain.CloudDisk
import com.magina.resource.domain.CloudDiskDTO
import com.magina.resource.domain.RequestParam
import com.google.common.collect.Lists
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
open class CloudDiskService {
	
	@Autowired
	private val cloudDiskDao: CloudDiskDAO? = null
	
	open fun listLinks(param: RequestParam) : List<CloudDisk> {
		if (cloudDiskDao != null) {
			return cloudDiskDao.listLinks(param)
		}
		return Lists.newArrayList()
	}
	
	open fun countLinks(query: String?) : Int {
		if (cloudDiskDao != null) {
			return cloudDiskDao.countLinks(query)
		}
		return 0
	}
	
	open fun getLinkById(linkId: Long) : CloudDisk {
		if (cloudDiskDao != null) {
			return cloudDiskDao.getLinkById(linkId)
		}
		return CloudDisk()
	}
	
	@Transactional
	open fun removeLinkById(linkId: Long) {
		if (cloudDiskDao != null) {
			cloudDiskDao.removeLinkById(linkId)
		}
	}
	
	@Transactional
	open fun saveLink(cloudDisk: CloudDiskDTO) {
		if (cloudDiskDao != null) {
			cloudDiskDao.saveLink(cloudDisk)
		}
	}
	
	@Transactional
	open fun updateLink(cloudDisk: CloudDiskDTO) {
		if (cloudDiskDao != null) {
			cloudDiskDao.updateLink(cloudDisk)
		}
	}
}